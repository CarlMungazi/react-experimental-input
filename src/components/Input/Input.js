import React, {Component} from 'react';
import PropTypes from 'prop-types';

/** A standard input component */
class Input extends Component {
  render() {
    const nativeAttrs = { ...this.props }

    return (
      <input {...nativeAttrs} />
    );
  }
}

Input.propTypes = {
  /**
   * The type of input 
   */
  type: PropTypes.string
}

Input.defaultProps = {
  type: "text"
}

export default Input;
