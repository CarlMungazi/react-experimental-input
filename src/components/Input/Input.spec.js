import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

test('it creates an input element', () => {
  const tree = renderer.create(
    <Input />
  ).toJSON();

  expect(tree).toMatchSnapshot();
})

test('it renders error messages', () => {
  const tree = renderer.create(
    <Input error='This is an error' />
  ).toJSON();
  
  expect(tree).toMatchSnapshot();
})