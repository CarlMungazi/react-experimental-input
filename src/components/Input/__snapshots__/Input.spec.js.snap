// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it creates an input element 1`] = `
<div>
  <input
    minLength=""
    onChange={[Function]}
    placeholder="Enter your text here"
    required={false}
    style={Object {}}
    type="text"
    value=""
  />
</div>
`;
